vembrane/modules/sort.py

```diff
@@ -0,0 +1,359 @@
+import heapq
+import math
+import sys
+import tempfile
+from functools import partial, total_ordering
+from itertools import batched
+from pathlib import Path
+from typing import Any, Callable, Dict, Iterable, List, Self, Sequence, Tuple, TypeVar
+
+from vembrane import __version__
+from vembrane.ann_types import NA
+from vembrane.backend.base import Backend, VCFReader, VCFRecord, VCFWriter
+from vembrane.common import (
+    add_common_arguments,
+    check_expression,
+    create_reader,
+    create_writer,
+    normalize,
+)
+from vembrane.errors import VembraneError
+from vembrane.representations import Annotation, SourceEnvironment
+
+
+def add_subcommand(subparsers):
+    parser = subparsers.add_parser(
+        "sort",
+        help="Sort VCF records by one or multiple Python expressions.",
+        description="Sort VCF records by one or multiple Python expressions that "
+        "encode keys for the desired order. This feature loads the entire VCF file "
+        "into memory in order to maximize performance. It is thus meant to sort small, "
+        "already filtered VCF files, e.g. for prioritizing records for the human eye. "
+        "For large VCF files, the only relevant sorting is usually by position, "
+        "which is better done with e.g. bcftools (and usually the sorting that variant "
+        "callers output).",
+    )
+    parser.add_argument(
+        "vcf",
+        help="The VCF/BCF file containing the variants. If not specified, "
+        "reads from STDIN.",
+        nargs="?",
+        default="-",
+    )
+    parser.add_argument(
+        "expression",
+        type=check_expression,
+        help="Python expression (or tuple of expressions) returning orderable values "
+        "to sort the VCF records by (ascending, smallest values coming first). "
+        "If multiple expressions are provided as a tuple, they are prioritized from "
+        "left to right with lowest priority on the right. "
+        "NA/NaN values are sorted to the end.",
+    )
+    parser.add_argument(
+        "--output",
+        "-o",
+        default="-",
+        help="Output file, if not specified, output is written to STDOUT.",
+    )
+    parser.add_argument(
+        "--output-fmt",
+        "-O",
+        default="vcf",
+        choices=["vcf", "bcf", "uncompressed-bcf"],
+        help="Output format.",
+    )
+    parser.add_argument(
+        "--preserve-annotation-order",
+        action="store_true",
+        help="If set, annotations are not sorted within the records, but kept in the "
+        "same order as in the input VCF file. If not set (default), annotations "
+        "are sorted within the record according to the given keys if any of the sort "
+        "keys given in the python expression refers to an annotation.",
+    )
+    parser.add_argument(
+        "--chunk-size",
+        type=int,
+        default=1000,
```

Comment:
1000 seems like a small number, I guess a single record only occupies a couple hundred bytes, so should be safe to raise to 10_000 if not even 100_000?